# **Exp 6**

# Aim: Understanding Support Vector Machine algorithm through building SVM algorithm in Python.


In [1]:
# Importing the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
# Importing the dataset
df = pd.read_csv("/content/drive/MyDrive/DA_datasets/Exp6/Social_Network_Ads.csv")
df

In [5]:
df.head()

User ID  Gender  Age  EstimatedSalary  Purchased
0  15624510    Male   19            19000          0
1  15810944    Male   35            20000          0
2  15668575  Female   26            43000          0
3  15603246  Female   27            57000          0
4  15804002    Male   19            76000          0

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   User ID          400 non-null    int64 
 1   Gender           400 non-null    object
 2   Age              400 non-null    int64 
 3   EstimatedSalary  400 non-null    int64 
 4   Purchased        400 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 15.8+ KB


In [7]:
X = df.iloc[:, [2,3]].values
Y = df.iloc[:, 4].values

In [8]:
# Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size = 0.25, random_state = 0)

In [9]:
from sklearn import svm
clf_svm = svm.SVC(kernel='poly')
clf_svm.fit(X_train, y_train)
y_pred = clf_svm.predict(X_test)
clf_svm.score(X_test, y_test) * 100

79.0

In [10]:
clf_svm.support_vectors_

array([[4.90e+01, 6.50e+04],
       [3.10e+01, 8.90e+04],
       [4.60e+01, 8.20e+04],
       [3.80e+01, 1.12e+05],
       [4.60e+01, 9.60e+04],
       [2.80e+01, 8.90e+04],
       [4.10e+01, 7.20e+04],
       [3.30e+01, 6.90e+04],
       [2.00e+01, 8.20e+04],
       [3.10e+01, 7.40e+04],
       [4.20e+01, 8.00e+04],
       [3.50e+01, 7.20e+04],
       [3.50e+01, 7.50e+04],
       [3.60e+01, 7.50e+04],
       [2.00e+01, 8.20e+04],
       [4.20e+01, 6.50e+04],
       [2.60e+01, 8.00e+04],
       [2.10e+01, 6.80e+04],
       [2.70e+01, 9.00e+04],
       [2.50e+01, 8.70e+04],
       [3.50e+01, 7.10e+04],
       [2.60e+01, 8.60e+04],
       [2.50e+01, 7.90e+04],
       [3.30e+01, 1.13e+05],
       [4.10e+01, 8.00e+04],
       [3.10e+01, 7.10e+04],
       [4.00e+01, 7.80e+04],
       [4.60e+01, 7.40e+04],
       [2.30e+01, 8.20e+04],
       [4.20e+01, 6.50e+04],
       [2.80e+01, 8.40e+04],
       [2.60e+01, 8.40e+04],
       [3.70e+01, 7.10e+04],
       [3.80e+01, 6.50e+04],
       [4.80e+

In [11]:
# Making the Confusion Matrix 
from sklearn import metrics
print(metrics.confusion_matrix(y_test, y_pred, labels=[0, 1]))

[[66  2]
 [19 13]]


In [12]:
print("="*60)
print(metrics.classification_report(y_test, y_pred, labels=[0, 1]))

              precision    recall  f1-score   support

           0       0.78      0.97      0.86        68
           1       0.87      0.41      0.55        32

    accuracy                           0.79       100
   macro avg       0.82      0.69      0.71       100
weighted avg       0.81      0.79      0.76       100



# Conclusion:
SVM  support-vector machines are supervised learning models with associated learning algorithms that analyze data for classification and regression analysis.The goal of the SVM algorithm is to create the best line or decision boundary that can segregate n-dimensional space into classes so that we can easily put the new data point in the correct category in the future. This best decision boundary is called a hyperplane.
while performing this experiment,I found that accurracy is 79%.On trying all the kernel methods, I found 'poly' as best kernel.
66 are True Positive, 13 are False Negative. 2 are False Positive whereas 16 are True Negative. 